src/PROPOSAL/PROPOSAL/crosssection/CrossSectionVector.h

```diff
@@ -13,29 +13,42 @@ struct CrossSectionVector {
 
     template <typename CrossVec>
     static std::vector<InteractionType> GetInteractionTypes(
-        CrossVec&& cross_vec)
+        CrossVec const& cross)
     {
         auto v = std::vector<InteractionType>();
-        for (auto& c_ptr : cross_vec)
-            v.push_back(c_ptr->GetInteractionType());
+        for (auto const& c : cross)
+            v.push_back(c->GetInteractionType());
         return v;
     }
 
-    template <typename CrossVec> static double GetLowerLim(CrossVec&& cross_vec)
+    template <typename CrossVec>
+    static double GetLowerLim(CrossVec const& cross)
     {
         using val_t = typename std::decay<CrossVec>::type::value_type;
-        auto result = std::max_element(
-            cross_vec.begin(), cross_vec.end(), [](val_t a_ptr, val_t b_ptr) {
+        auto r = std::max_element(
+            cross.begin(), cross.end(), [](val_t a_ptr, val_t b_ptr) {
                 return a_ptr->GetLowerEnergyLim() > b_ptr->GetLowerEnergyLim();
             });
-        return (*result)->GetLowerEnergyLim();
+        return (*r)->GetLowerEnergyLim();
+    }
+
+    template <typename CrossVec>
+    static double GetMinStochasticEnergy(CrossVec const& cross)
+    {
+        using val_t = typename std::decay<CrossVec>::type::value_type;
+        auto r = std::max_element(
```

Comment:
Why is this the maximum instead of the minimum?

Reply:
Wait, this is `std::max_element`, but the comparison function object is defined the wrong way around, to it essentially is `std::min_element`, isn't it? 😅

Reply:
But this does not exist in c++14. It is a `c++17` feature :see_no_evil:

Reply:
But max_element as well. https://en.cppreference.com/w/cpp/algorithm/max_element

Reply:
No, min_element exists also since basically forever: https://en.cppreference.com/w/cpp/algorithm/min_element

Reply:
UFF, I have overseen the until. Good catch :+1: